server/database/config/db_build.sql

```diff
@@ -101,7 +101,7 @@ CREATE TABLE filter (
 
 CREATE TABLE saved_offer (
   member_id INTEGER REFERENCES member(id),
-  offet_id INTEGER REFERENCES offer(id)
+  offer_id INTEGER REFERENCES offer(id) ON DELETE CASCADE
```

Comment:
@Angham116 This is probably out of scope for this project, but sometimes it can help the user experience if your app doesn't actually delete records, but instead sets a field to indicate that it is deleted (for example the offer's status could be "deleted") but keeps the data in the table.
Then it's a little more work to filter out deleted offers, but the benefit is that you could explain to a user what happened when their saved offer is deleted.  Instead of the user wondering why their saved offer disappeared, the app could tell the user that the offer they had saved has been removed or expired.  In other words, you can make the experience less confusing when someone else deletes data that a person expects to see.
It's something you all could try for the next app if you'd like :)